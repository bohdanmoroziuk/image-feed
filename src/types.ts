export interface Item {
  id: string;
  author: string;
}

export interface Comment {
  id: string;
  text: string;
}
