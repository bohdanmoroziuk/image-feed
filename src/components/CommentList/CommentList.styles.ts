import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  comment: {
    marginLeft: 20,
    paddingVertical: 20,
    paddingRight: 20,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: 'rgba(0, 0, 0, 0.05)',
  },
});
