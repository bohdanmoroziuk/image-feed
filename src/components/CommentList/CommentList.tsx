import { FunctionComponent } from 'react';
import { View, Text, ScrollView } from 'react-native';

import { Comment } from 'src/types';

import styles from './CommentList.styles';

export interface CommentListProps {
  items: Comment[];
}

const renderItem = (item: Comment) => (
  <View
    style={styles.comment}
    key={item.id}
  >
    <Text>
      {item.text}
    </Text>
  </View>
);

export const CommentList: FunctionComponent<CommentListProps> = ({ items }) => (
  <ScrollView>
    {items.map(renderItem)}
  </ScrollView>
);
