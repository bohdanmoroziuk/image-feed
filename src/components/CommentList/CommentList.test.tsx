import { render } from '@testing-library/react-native';
import uuid from 'react-native-uuid';

import { Comment } from 'src/types';
import { CommentList } from './CommentList';

const mockComments: Comment[] = [
  {
    id: uuid.v4().toString(),
    text: 'Awesome',
  },
  {
    id: uuid.v4().toString(),
    text: 'Nothing special',
  },
];

describe('CommentList', () => {
  it('renders successfully', () => {
    const { toJSON } = render(
      <CommentList items={mockComments} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { getByText } = render(
      <CommentList items={mockComments} />
    );

    expect(getByText(mockComments[0].text)).toBeDefined();
    expect(getByText(mockComments[1].text)).toBeDefined();
  });
});
