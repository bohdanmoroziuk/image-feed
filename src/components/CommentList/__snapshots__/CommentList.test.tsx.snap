// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList renders successfully 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(0, 0, 0, 0.05)",
          "borderBottomWidth": 0.5,
          "marginLeft": 20,
          "paddingRight": 20,
          "paddingVertical": 20,
        }
      }
    >
      <Text>
        Awesome
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(0, 0, 0, 0.05)",
          "borderBottomWidth": 0.5,
          "marginLeft": 20,
          "paddingRight": 20,
          "paddingVertical": 20,
        }
      }
    >
      <Text>
        Nothing special
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
